### PR TITLE
chore: use tmpdir

### PR DESCRIPTION
**Pull request by cisse21**
# Description

< Replace with adequate description for this PR as per [Pull Request document](https://www.notion.so/rudderstacks/Pull-Requests-40a4c6bd7a5e4387ba9029bab297c9e3) >

## Notion Ticket

< Replace with Notion Link >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: enterprise/replay/dumpsloader.go [hunk 1]
```diff
@@ -258,6 +271,12 @@ func (handle *dumpsLoaderHandleT) Setup(ctx context.Context, db *jobsdb.HandleT,
 	if err != nil {
 		panic("invalid start time format provided")
 	}
+
+	tmpDirPath, err := misc.CreateTMPDIR()
```
Comment (lvrach) [inline enterprise/replay/dumpsloader.go:287]: I was trying to understand if tmp dir is persist across pod restarts, then I saw we are using it for badger db. In prod we setting it in `RUDDER_TMPDIR=/data/rudderstack`. 

TL;DR: TMPDIR is a confusing name

**Review by lvrach — APPROVED** [review]

**Review by achettyiitr — APPROVED** [review]
